onnxruntime/core/providers/shared/initializer_view/initializer_view.h

```diff
@@ -0,0 +1,52 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include <memory>
+
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/tensor_shape.h"
+
+#include "gsl/gsl-lite.hpp"
+
+namespace onnxruntime {
+
+class InitializerView {
+ private:
+  InitializerView(const InitializerView&) = delete;
+  InitializerView& operator=(const InitializerView&) = delete;
+  InitializerView() = default;
+
+ public:
+  static std::unique_ptr<InitializerView> Create(
+      const ONNX_NAMESPACE::TensorProto& tensor_proto);
+  const TensorShape& Shape() {
+    return shape_;
+  }
+
+  const DataTypeImpl* Type() {
+    return type_;
+  }
+
+  template <class T>
+  gsl::span<const T> DataAsSpan() {
+    ORT_ENFORCE(unpacked_tensor_.size() >= sizeof(T), " data bytes must more than sizeof (T)");
+    return gsl::make_span(data<T>(), unpacked_tensor_.size() / sizeof(T));
+  }
+
+ private:
+  template <class T>
+  const T* data() {
+    // make sure T has no qualifier
+    static_assert(std::is_same_v<typename std::decay<T>::type, T> &&
+                  std::is_same_v<typename std::remove_pointer<T>::type, T>);
+    return reinterpret_cast<const T*>(unpacked_tensor_.data());
+  }
+
+ private:
+  TensorShape shape_;
+  const DataTypeImpl* type_{nullptr};
```

Comment:
>const DataTypeImpl* type_{nullptr}; [](http://example.com/codeflow?start=2&length=35)

Would storing an enum be sufficient? What other information is needed here from DataTypeImpl?

Reply:
`Initializer`   deserialized `TensorProto` to `Tensor`, then users can easily get its shape and type. 
Here we take vector<uint8> as data structure to hold data, Type and Shape would be lost. I have to add two attributes to hold it.